### PR TITLE
allow all batch jobs to be scheduled on preemptibles

### PR DESCRIPTION
**Pull request by cseed**
I think this is a good change, but we should keep in eye out, it is going to increase preemptions and restarts, and will almost certainly expose issues.  Poor man's chaos.

I will add an option to the batch interface to disable this for longer-running jobs.

### CODE REVIEW

File: batch/batch/server/server.py [hunk 1]
```diff
@@ -620,6 +623,10 @@ def to_dict(self):
     if 'memory' not in pod_spec.containers[0].resources.requests:
         pod_spec.containers[0].resources.requests['memory'] = '500M'
 
+    if not pod_spec.tolerations:
+        pod_spec.tolerations = []
+    pod_spec.tolerations.extend(kube.client.V1Toleration(key='preemptible', value='true'))
```
Comment (akotlar) [inline batch/batch/server/server.py]: I think we want pod_spec.tolerations to be a [V1Toleration], so I believe .append is more appropriate (or just literal `pod_spec.tolerations = [ kube.client.V1Toleration(key='preemptible', value='true') ]`

Comment (cseed) [reply batch/batch/server/server.py]: The client may have already specified other tolerations.

Comment (akotlar) [reply batch/batch/server/server.py]: Ah, so in this case `kube.client.V1Toleration` doesn't return a single V1Toleration but a list of them?

Comment (cseed) [reply batch/batch/server/server.py]: Sorry, you're right, I needed `append`.  Fixed.

**Pull request merged**